Update and improve gVisor support:

jcpittman144 opened the pull request:
- Implement event handlers for recently-added gVisor native events, including WRITE
- Enhance event handlers to explicitly document which SCAP event fields are invalid (e.g. where info is not provided by the native gVisor event)
- Improve gvisor config generator, to configure all fields needed by the event parser, and only those fields
- Zero fill allocated scap_gvisor_platform structure, to avoid crash on startup
- Return SCAP_TIMEOUT instead of SCAP_EOF on interrupted socket read
- Add new gvisor parser unit tests for new event handlers; and fix existing unit tests to reflect updated functionality of existing event handlers

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines in the [CONTRIBUTING.md](https://github.com/falcosecurity/.github/blob/master/CONTRIBUTING.md) file and learn how to compile Falco from source [here](https://falco.org/docs/source).
2. Please label this pull request according to what type of issue you are addressing.
3. Please add a release note!
4. If the PR is unfinished while opening it specify a wip in the title before the actual title, for example, "wip: my awesome feature"
-->

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

<!--
Please remove the leading whitespace before the `/kind <>` you uncommented.
-->

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

/area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap-engine-udig

> /area libscap

> /area libpman

/area libsinsp

/area tests

> /area proposals

<!--
Please remove the leading whitespace before the `/area <>` you uncommented.
-->

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

<!--
Please remove the leading whitespace before the `/version <>` you uncommented.
-->

**What this PR does / why we need it**:
Enhance and update gVisor support

**Which issue(s) this PR fixes**:

<!--
Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
If PR is `kind/failing-tests` please post the related issues/tests in a comment and do not use `Fixes`.
-->

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

<!--
If no, you have to do nothing.
If yes, a release note is required:
Delete `NONE` and enter your extended release note in the block below.
Please note, the release note follows the "conventional commit specification" (https://www.conventionalcommits.org/en/v1.0.0/):
For example: `fix: broken link`.
If the PR requires additional action from users switching to the new release, prepend the string "action required:".
For example, `action required: change the API interface of libscap`.
-->

```release-note
fix: Enhance and update gVisor support, including additional system call events
```

FedeDP commented:
/cc @LucaGuerra @loresuso that were original authors of gvisor  :)

dwindsor commented on userspace/libscap/engine/gvisor/parsers.cpp:
```diff
@@ -143,28 +171,23 @@ static parse_result parse_container_start(const char *proto, size_t proto_size,
+	                 tid_field,  // tid
+	                 tgid_field, // pid
+	                 1,          // ptid for initial process
+	                 "/",        // cwd for initial process
```
Nit: this is actually different from the `cwd` value (`""`)  in the previous call to `scap_event_encode_params`, which also IIUC encodes process data for `init`.

jcpittman144 replied:
Good catch.
I thought I was doing the right thing, by hardcoding the current working directory as "/", for the container start event.  But now I'm not so sure.
I will restore the original value (empty string).

LucaGuerra commented:
This is massive! Thanks for adding tests, these are really important if we ever want to refactor any of it since it's a lot of code :pray: .
Re. the "filler" structure we can see if it's the right thing for us and change it later. We definitely want these improvements in the next version of libs. Please take a look at @dwindsor 's comment above but I can already approve this. Thanks!

poiana commented:
LGTM label has been added.  <details>Git tree hash: 6c1687c6d6d4f06b6308265aa66ef5a9b354fce3</details>

LucaGuerra commented:
/hold

@jcpittman144 once you address review comments please simply add a message with slash unhold to merge the PR

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/libs/pull/1394#" title="Author self-approved">jcpittman144</a>*, *<a href="https://github.com/falcosecurity/libs/pull/1394#pullrequestreview-1673484964" title="Approved">LucaGuerra</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Flibs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/libs/blob/master/OWNERS)~~ [LucaGuerra]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/libs/pull/1394#" title="Author self-approved">jcpittman144</a>*
**Once this PR has been reviewed and has the lgtm label**, please ask for approval from lucaguerra. For more information see [the Kubernetes Code Review Process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process).

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Flibs).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/falcosecurity/libs/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["lucaguerra"]} -->

poiana commented:
New changes are detected. LGTM label has been removed.

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/libs/pull/1394#" title="Author self-approved">jcpittman144</a>*
**Once this PR has been reviewed and has the lgtm label**, please ask for approval from lucaguerra. For more information see [the Kubernetes Code Review Process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process).

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Flibs).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/falcosecurity/libs/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["lucaguerra"]} -->

jcpittman144 commented:
/unhold

jcpittman144 commented:
@LucaGuerra, @dwindsor,
I addressed the review comment, and made a change to restore the original behavior of one of the generated events.
That change invalidated the LGTM designation.
So the PR is blocked again, waiting for an approval.
If everything looks good, please approve and I will do the merge.
Thanks!

LucaGuerra reviewed (approved):
Thank you!

poiana commented:
LGTM label has been added.  <details>Git tree hash: becd1c9512da1ce19c3992689378c9719942e01c</details>

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/libs/pull/1394#" title="Author self-approved">jcpittman144</a>*, *<a href="https://github.com/falcosecurity/libs/pull/1394#pullrequestreview-1679697149" title="Approved">LucaGuerra</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Flibs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/libs/blob/master/OWNERS)~~ [LucaGuerra]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/libs/pull/1394#" title="Author self-approved">jcpittman144</a>*, *<a href="https://github.com/falcosecurity/libs/pull/1394#pullrequestreview-1681972525" title="Approved">leogr</a>*, *<a href="https://github.com/falcosecurity/libs/pull/1394#pullrequestreview-1679697149" title="Approved">LucaGuerra</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Flibs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/libs/blob/master/OWNERS)~~ [LucaGuerra,leogr]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->
